lib/workflows/send_errored_vm_email.py

```diff
@@ -0,0 +1,180 @@
+
+
+# pylint:disable=too-many-arguments
+# pylint:disable=too-many-locals
+def send_errored_vm_email(
+    smtp_account: SMTPAccount,
+    cloud_account: Union[CloudDomains, str],
+    limit_by_projects: Optional[List[str]] = None,
+    time_variable: int = -1,
+    all_projects: bool = False,
+    as_html: bool = False,
+    send_email: bool = False,
+    use_override: bool = False,
+    override_email_address: Optional[str] = "cloud-support@stfc.ac.uk",
+    cc_cloud_support: bool = False,
+    **email_params_kwargs,
+):
+    """
+    Sends an email to each user who owns one or more VMs that are in error state
+    This email will contain a notice to delete or rebuild the VM
+    :param smtp_account: (SMTPAccount): SMTP config
+    :param cloud_account: string represents cloud account to use
+    :param limit_by_projects: A list of project names or ids to limit search in
+    :param all_projects: A boolean which if selected will search in all projects
+    :param send_email: Actually send the email instead of printing what will be sent
+    :param as_html: Send email as html
+    :param use_override: flag if set will use override email address
+    :param override_email_address: an overriding email address to use if override_email set
+    :param cc_cloud_support: flag if set will cc cloud-support email address to each generated email
+    :param email_params_kwargs: see EmailParams dataclass class docstring
+    """
+    if limit_by_projects and all_projects:
+        raise RuntimeError(
+            "given both project list and all_projects flag - please choose only one"
+        )
+    if not (limit_by_projects or all_projects):
+        raise RuntimeError(
+            "please provide either a list of project identifiers or with flag 'all_projects' to run globally"
+        )
+
+    server_query = find_servers_with_errored_vms(cloud_account, time_variable, limit_by_projects)
+
+    for user_id in server_query.to_props().keys():
+        user_name, email_addr = find_user_info(
+            user_id, cloud_account, override_email_address
+        )
+        send_to = [email_addr]
+        if use_override:
+            send_to = [override_email_address]
+
+        if as_html:
+            server_list = server_query.to_html(
+                groups=[user_id], include_group_titles=False
+            )
+        else:
+            server_list = server_query.to_string(
+                groups=[user_id], include_group_titles=False
+            )
+
+        if not send_email:
+            print_email_params(send_to[0], user_name, as_html, server_list)
+
+        else:
+            email_params = build_email_params(
+                user_name,
+                server_list,
+                email_to=send_to,
+                as_html=as_html,
+                email_cc=("cloud-support@stfc.ac.uk",) if cc_cloud_support else None,
+                **email_params_kwargs,
+            )
+            Emailer(smtp_account).send_emails([email_params])
```

Comment:
NIT: remove the else by returning early
```suggestion
        if not send_email:
            print_email_params(send_to[0], user_name, as_html, server_list)
            return 
            
        email_params = build_email_params(
            user_name,
            server_list,
            email_to=send_to,
            as_html=as_html,
            email_cc=("cloud-support@stfc.ac.uk",) if cc_cloud_support else None,
            **email_params_kwargs,
            )
            Emailer(smtp_account).send_emails([email_params])
```